kasa/httpclient.py

```diff
@@ -117,6 +117,21 @@ async def post(
                     response_data = await resp.read()
                     if return_json:
                         response_data = json_loads(response_data.decode())
+                else:
+                    response_data = await resp.read()
+                    _LOGGER.debug(
+                        "Device %s received status code %s with response %s",
+                        self._config.host,
+                        resp.status,
+                        str(response_data),
+                    )
+                    if response_data and return_json:
+                        try:
+                            response_data = json_loads(response_data.decode())
+                        except Exception:
+                            _LOGGER.debug(
+                                "Device %s response could not be parsed as json"
+                            )
```

Comment:
Should this raise or return?

Reply:
It will still return the `bytes` which is one of the return types for response. It's up to the caller to then decide whether to raise or not.